provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt

```diff
@@ -200,7 +201,9 @@ open class PactBrokerLoader(
 
       if (pactBrokerConsumers.isNotEmpty()) {
         val consumerInclusions = pactBrokerConsumers.flatMap { parseListExpression(it, resolver) }
-        consumers = consumers.filter { consumerInclusions.isEmpty() || consumerInclusions.contains(it.name) }
+        if (consumerInclusions.isNotEmpty()) {
+          consumers = consumers.filter { consumer -> consumerInclusions.any { inclusion -> consumer.name.contains(inclusion) } }
```

Comment:
The risk with this is that there are a lot of words in the name that could contain the specified consumer name. Right now, all results are being filtered out, but this could make it match so not enough are being filtered out.

As was pointed out in the issue, `consumer.name` here will be something like
> Pact between consumer (whatever-the-consumer-version-is) and provider

A contrived example would be if I have a provider (my-provider) that serves two consumers (my-consumer and my-consumer-legacy)

> Pact between my-consumer-legacy (123abc) and my-provider
> Pact between my-consumer (456def) and my-provider

In this case, if you had the consumer property set to `my-consumer`, it would still run verifications for `my-consumer-legacy`

Reply:
I agree and thought about that, decided to discuss it here.

As I see this endpoint `/pacts/provider/billing/latest` returns:
```json
"pb:pacts": [
            {
                "href": "https://pact-broker.everon.dev/pacts/provider/billing/consumer/dataflow/version/1231f00fea1c41259472b9be79fe69bca0bd178a",
                "title": "Pact between dataflow (1231f00fea1c41259472b9be79fe69bca0bd178a) and billing",
                "name": "dataflow"
            }
        ]
```
And it is okay, we can take the name from there

but this `pacts/provider/billing/for-verification` returns:
```json
"_links": {
                    "self": {
                        "href": "https://pact-broker.everon.dev/pacts/provider/billing/consumer/dataflow/pact-version/af8018d15849cd333429faa7404171d2b4ab588d",
                        "name": "Pact between dataflow (1231f00fea1c41259472b9be79fe69bca0bd178a) and billing"
                    }
                }
```
I see two options:

1. if pact broker will align `pacts/provider/billing/for-verification` with `/pacts/provider/billing/latest` then it will be okay and no complicated filters will be needed, because we'll have `name` and `title`.

2. `pact-jvm` changes the logic of using endpoints and here I'm not aware of the reasoning behind it

I'd made a rush maybe, but now I see this is a better solution, please consider @pendsley @anto-ac
If you see that it should be changed on Pact broker side I'd appreciate some reference in this task to PactBroker one

Reply:
I can add the consumer and provider names into the 'pacts for verification' response. I didn't realise anyone was filtering on the title!

Reply:
it uses `name`, but at `/for-verification` name value looks like _title_ and the `title` is missing

I feel like we need pacts for pact client libs :)

Reply:
Indeed. Unfortunately, this particular bug arises not because of a format misunderstanding between pact-jvm and the Pact Broker, but because there is a change in where the responsibility lies for filtering by consumer. 

For the old API, the client was doing it. For the new API, the Pact Broker is doing it.

The correct thing to do with the consumer name filter is to pass the name to the Pact Broker via the selectors, and filter on the server side, as this is now supported.

The problem is that there are going to be some people who are still going to be using the old API to fetch the pacts. The consumer filter should only be applied client side if the old API was used to fetch the pacts. That is going to have to be passed in to the filtering code somehow.